src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size;
+    private int index;
+    private final Object[] array1 = new Object[DEFAULT_SIZE];
+    private T[] values = (T[]) array1;
+
     @Override
     public void add(T value) {
-
+        if (size >= values.length) {
+            values = Arrays.copyOf(values, (int) (values.length * 1.5));
+        }
```

Comment:
create method grow()
Instead of 1.5, use a constant
[You need to use System.arraycopy()](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#use-systemarraycopy-to-move-your-array-elements). Fix it in all places
extract logic that inside if to grow() method. Reuse it everywhere